modules/router-store/src/router_store_module.ts

```diff
@@ -302,7 +300,13 @@ export class StoreRouterConnectingModule {
   private dispatchRouterAction(type: string, payload: any): void {
     this.trigger = RouterTrigger.ROUTER;
     try {
-      this.store.dispatch({ type, payload });
+      this.store.dispatch({
+        type,
+        payload: {
+          routerState: this.routerState,
```

Comment:
The `routerState` dispatched during the `NavigationEnd` event is not the same as the `NavigationStart`. The router state at that point is fully formed with resolved data and everything else, so we would need to capture that before dispatching that action.